### PR TITLE
Update Dependabot Setup

### PR DESCRIPTION
**Pull request by wangerekaharun**
### **User description**
Story: https://app.shortcut.com/smileid/story/15609
## Summary

Setup Dependabot

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Other

___

### **Description**
• Consolidate Dependabot configuration into single YAML file
• Add Gradle package ecosystem with Android plugin grouping
• Standardize timezone and schedule formatting across ecosystems
• Remove duplicate pub ecosystem configuration file

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>dependabot.yaml</strong><dd><code>Remove duplicate Dependabot configuration file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/dependabot.yaml

• Remove entire file containing pub package ecosystem configuration

</details>

  </td>
  <td><a href="https://github.com/smileidentity/flutter/pull/179/files#diff-ee0c08340ace9ac3ce3c78e2377968bca2376b8bfd9a7a48d96515f5490a902c">+0/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>dependabot.yml</strong><dd><code>Enhance Dependabot with multiple package ecosystems</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/dependabot.yml

• Add comprehensive header comment with documentation link<br> • <br>Standardize GitHub Actions configuration with timezone and time<br> • Add <br>pub package ecosystem with weekly schedule<br> • Add Gradle ecosystem with <br>Android plugin grouping

</details>

  </td>
  <td><a href="https://github.com/smileidentity/flutter/pull/179/files#diff-dd4fbda47e51f1e35defb9275a9cd9c212ecde0b870cba89ddaaae65c5f3cd28">+32/-10</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
-  groups:
-    github-actions:
-      patterns:
-      - '*'
+  - package-ecosystem: github-actions
+    directory: "/"
+    schedule:
+      interval: monthly
+      time: "10:00"
+      timezone: "America/Los_Angeles"
```
Comment (robin-smileid) [inline .github/dependabot.yml:9]: Should we change this timezone to something more aligned with the location the mobile team is in? London/Kenya/ ...

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
-    day: monday
-    time: "13:00"
-  groups:
-    github-actions:
-      patterns:
-      - '*'
+  - package-ecosystem: github-actions
+    directory: "/"
+    schedule:
+      interval: monthly
```
Comment (jumaallan) [inline .github/dependabot.yml:7]: move to weekly too

**Review by jumaallan — APPROVED** [review]
